packages/ListBox/src/ListBox.js

```diff
@@ -0,0 +1,164 @@
+  const { children, height, placeholder, Trigger: TriggerProps, Footer, Filter } = props;
+
+  const handleEffectHasFooter = effects.handleEffectHasFooter(Footer, dispatch);
+  React.useEffect(handleEffectHasFooter, []);
+
+  const triggerProps = {
+    hasClearButton: true,
+    onClickClear: null,
+    placeholder,
+    onFooterClickAccept: Footer ? Footer.props.onClickAccept : null,
+  };
+
+  const trigger = TriggerProps ? (
+    React.cloneElement(TriggerProps, { ...triggerProps, ...TriggerProps.props })
+  ) : (
+    <Trigger {...triggerProps} />
+  );
+
+  return (
+    <React.Fragment>
+      {trigger}
+      <Content>
+        <Box>
+          {Filter}
+          <List height={height}>
+            <Options>{children}</Options>
+          </List>
+          {Filter ? <NoResults label={Filter.props.noResultsMessage} /> : null}
+          {Footer ? React.cloneElement(Footer, { ref: state.refFooterContainer }) : null}
+        </Box>
+      </Content>
+    </React.Fragment>
+  );
+}
+
+ListBox.propTypes = {
+  ...propTypes,
+  children: PropTypes.node.isRequired,
+  height: PropTypes.number.isRequired,
+  placeholder: PropTypes.string.isRequired,
+};
+
+ListBox.defaultProps = {};
+
+const ListBoxContainer = React.forwardRef((props, ref) => {
+  const [state, dispatch] = useListBox();
+
+  const {
+    children,
+    height,
+    isInline,
+    placeholder,
+    Filter, // eslint-disable-line
+    Trigger, // eslint-disable-line
+    Footer, // eslint-disable-line
+    Popover: PopoverWithProps, // eslint-disable-line
+  } = props;
+
+  const shouldTriggerKeepFocus = PopoverWithProps && PopoverWithProps.props.shouldKeepFocus;
+
+  // IMPERATIVE API
+  const imperativeHandle = handleImperative(state, dispatch);
+  React.useImperativeHandle(ref, imperativeHandle);
+
+  // EFFECTS
+  const handleEffectChildren = effects.handleEffectChildren(props, state, dispatch);
+  const handleEffectIsDisabledChange = effects.handleEffectIsDisabledChange(props, dispatch);
+  const handleEffectIsPopOverOpen = effects.handleEffectIsPopOverOpen(state, dispatch, shouldTriggerKeepFocus);
+  const handleEffectListBoxScrolled = effects.handleEffectListBoxScrolled(state);
+  const handleEffectListBoxWidth = effects.handleEffectListBoxWidth(state, dispatch);
+  const handleEffectOptionSelected = effects.handleEffectOptionSelected(state, dispatch);
+
+  React.useEffect(handleEffectIsDisabledChange, [props.isDisabled]);
+  React.useEffect(handleEffectListBoxWidth, [state.refTriggerContainer.current]);
+  React.useEffect(handleEffectOptionSelected, [state.selectedOptions]);
+  React.useEffect(handleEffectChildren, [props.children]);
+  React.useLayoutEffect(handleEffectIsPopOverOpen, [state.isOpen]);
+  React.useLayoutEffect(handleEffectListBoxScrolled, [state.activeOption]);
+
+  const ListBoxProps = {
+    children,
+    Filter,
+    Footer,
+    height,
+    placeholder,
+    Trigger,
+  };
+
+  const listBox = <ListBox {...ListBoxProps}>{children}</ListBox>;
```

Comment:
`listBox` is a component, so I think it should have a capital first letter.  `ListBox` is already being used though, so I'm not sure what else to call it... 😬